### PR TITLE
cli: create `jj file list` and deprecate `jj files`

### PR DESCRIPTION
**Pull request by fowles**
# Checklist

If applicable:
- [x] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [x] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/commands/file/mod.rs [hunk 1]
```diff
@@ -24,6 +25,8 @@ use crate::ui::Ui;
 pub enum FileCommand {
     Print(print::PrintArgs),
     Chmod(chmod::ChmodArgs),
+    #[command(visible_alias("ls"))]
```
Comment (PhilipMetzger) [inline cli/src/commands/file/mod.rs:28]: We need to come to some sort of consensus what core aliases are, see #3218 and https://github.com/martinvonz/jj/pull/3242#pullrequestreview-1923133220. cc @martinvonz @thoughtpolice

File: cli/src/commands/mod.rs [hunk 2]
```diff
@@ -94,7 +93,8 @@ enum Command {
     Edit(edit::EditArgs),
     #[command(subcommand)]
     File(file::FileCommand),
-    Files(files::FilesArgs),
+    #[command(hide = true)]
```
Comment (PhilipMetzger) [inline cli/src/commands/mod.rs:96]: Nit: Needs a deprecation comment and should not be hidden yet.

**Review by PhilipMetzger — CHANGES_REQUESTED** [review]
While this is mostly LG by itself, I'm going to block on the whole aliasing discussion. And I would prefer if you'd used the `cli:` prefix to make the affected component visible.

File: cli/src/commands/file/mod.rs (hunk 1 above)
Comment (fowles) [reply cli/src/commands/file/mod.rs:28]: Ok, I will simply remove the alias for now

File: cli/src/commands/mod.rs (hunk 2 above)
Comment (fowles) [reply cli/src/commands/mod.rs:96]: all the other deprecated things I see here are marked hidden.  Not sure what the pattern to follow is.

Comment (PhilipMetzger) [reply cli/src/commands/mod.rs:97]: I am dumb, you can keep it as is.

**Review by PhilipMetzger — APPROVED** [review]

**Pull request merged**